### DISCUSSION TITLE
Redirect to a custom setting in settings.json editor using command:workbench.action.openSettingsJson

### DISCUSSION THREAD

**Discussion by abhi4039** [microsoft/vscode-discussions#489] [Extension Development QnA] [answered]
Summary: 

I have a defined markdownDescription for a custom setting: "Some Description, [Edit in settings.json](command:workbench.action.openSettingsJson)",

On click of the link, I want the behavior of opening settings.json and placing the cursor to the linked setting.

Can someone suggest a way, I could acheive this behavior.

**Comment by alefragnani** [comment, marked as answer]
Hi @abhi4039 ,

As far as I could test, the `workbench.action.openSettingsJson` command doesn’t _focus_ the line where the setting you passed as parameter is located. Instead, it just opens the `settings.json` file, and from there, VS Code itself will restore the previous selection/position the cursor/viewport was in your latest session.

If this is true, I suggest you to open a feature request in VS Code repo (https://github.com/microsoft/vscode/issues/new/choose), and ask for such update in its behavior. 

In the meantime, you could focus the line by yourself. To accomplish this, you must first locate the text inside the `settings.json` editor which is opened and then, use the `vscode.window.activeTextEditor.selection`  and `vscode.window.activeTextEditor.revealRange` APIs to select and focus the line.

Hope this helps
